src/test/java/io/github/christophermanahan/carnitas/HTTPResponseTest.java

```diff
@@ -5,52 +5,108 @@
+    }
+
+    @Test
+    void itHasABody() {
+        Optional<String> body = Optional.of("name=<something>");
+
+        HTTPResponse response = new ResponseBuilder()
+          .setBody(body)
+          .get();
+
+        assertEquals(body, response.body());
+    }
+
+    @Test
+    void itCanTestEquality() {
+        HTTPResponse.Status status = HTTPResponse.Status.OK;
+        String header = Headers.contentLength(0);
+        Optional<String> body = Optional.of("name=<something>");
+        HTTPResponse response = new ResponseBuilder()
+          .setStatus(status)
+          .addHeader(header)
+          .setBody(body)
+          .get();
+        HTTPResponse otherResponse = new ResponseBuilder()
+          .setStatus(status)
+          .addHeader(header)
+          .setBody(body)
+          .get();
+
+        boolean equal = response.equals(otherResponse);
+
+        assertTrue(equal);
     }
 
     @Test
-    void itSerializesToAFormattedHTTPResponseWithBody() {
-        HTTPResponse.Status status = HTTPResponse.Status.CREATED;
-        String body = "name=<something>";
-        HTTPResponse httpResponse = new HTTPResponse(status)
-          .withBody(Optional.of(body));
-
-        String response = new String(httpResponse.serialize());
-
-        String expectedResponse = HTTPResponse.VERSION +  " " + status.code + HTTPResponse.CRLF
-          + Headers.contentLength(body.length())
-          + HTTPResponse.BLANK_LINE
-          + body;
-        assertEquals(expectedResponse, response);
+    void itCanTestInequalityOfStatus() {
+        HTTPResponse response = new ResponseBuilder()
+          .setStatus(HTTPResponse.Status.OK)
+          .get();
+        HTTPResponse otherResponse = new ResponseBuilder()
+          .setStatus(HTTPResponse.Status.CREATED)
+          .get();
+
+        boolean equal = response.equals(otherResponse);
+
+        assertFalse(equal);
+    }
+
+    @Test
+    void itCanTestInequalityOfHeaders() {
+        HTTPResponse.Status status = HTTPResponse.Status.OK;
+        HTTPResponse response = new ResponseBuilder()
+          .setStatus(status)
+          .addHeader(Headers.contentLength(0))
+          .get();
+        HTTPResponse otherResponse = new ResponseBuilder()
+          .setStatus(status)
+          .addHeader(Headers.contentLength(1))
+          .get();
+
+        boolean equal = response.equals(otherResponse);
+        assertFalse(equal);
+    }
+
+    @Test
+    void itCanTestInequalityOfBody() {
+        HTTPResponse.Status status = HTTPResponse.Status.OK;
+        HTTPResponse response = new ResponseBuilder()
+          .setStatus(status)
+          .setBody(Optional.of("name=<something>"))
+          .get();
+        HTTPResponse otherResponse = new ResponseBuilder()
+          .setStatus(status)
+          .setBody(Optional.of("name=<something-else>"))
+          .get();
+
+        boolean equal = response.equals(otherResponse);
+
+        assertFalse(equal);
```

Comment:
I like to use the `assertEquals` method for testing equality methods. It can provide a more informative error message, and the assertion also matches the test description a little better.

P.S. You can enhance the error message too, by adding a `toString` method.  I like to add them to data structures just for debugging purposes.

Reply:
Ah, I did not consider the fact that overriding equals would allow asserEquals to function with the new method! The `toString` method makes a lot of sense in the same way.